cloud/blockstore/libs/rdma/impl/client.cpp

```diff
@@ -952,7 +951,21 @@ void TClientEndpoint::SendRequest(TRequestPtr req, TSendWr* send)
     requestMsg->Out = req->OutBuffer;
 
     RDMA_TRACE("SEND " << TWorkRequestId(send->wr.wr_id));
-    Verbs->PostSend(Connection->qp, &send->wr);
+
+    try {
+        Verbs->PostSend(Connection->qp, &send->wr);
```

Comment:
lest's call `SendRequestCompleted(send, IBV_WC_FATAL_ERR)` if PostSend fails. This way we already invoke our handler for failed SendRequest and implicitly invoke `Disconnect`

Same thing with `PostRecv` where we can call `RecvResponseCompleted` with error status.

Reply:
I'd rather not. The way counters are calculated right now, it will lead to double decrement